synapse/rest/client/v2_alpha/auth.py

```diff
@@ -90,6 +89,30 @@
 </html>
 """
 
+SUCCESS_TEMPLATE = """
+<html>
+<head>
+<title>Success!</title>
+<meta name='viewport' content='width=device-width, initial-scale=1,
+    user-scalable=no, minimum-scale=1.0, maximum-scale=1.0'>
+<link rel="stylesheet" href="/_matrix/static/client/register/style.css">
+<script>
+if (window.onAuthDone) {
+    window.onAuthDone();
+} else if (window.opener && window.opener.postMessage) {
+     window.opener.postMessage("authDone", "*");
+}
+</script>
```

Comment:
I'm not realizing that this template was probably never correct to use due to this code anyway. It didn't really hurt, but was unnecessary, I think.

Reply:
Hmm...actually maybe this is necessary since we do go through the fallback auth in this situation.

Reply:
Might be worth asking a client dev about it.

Reply:
@turt2live Any idea about this?

Reply:
Re-reading the spec ... it is definitely required. The examples in the MSC even give it! Doh! 😕

Reply:
I re-added this.